admiral/stage.py

```diff
@@ -56,6 +57,11 @@ def run_analysis(full_trained_directory, full_subscript, parameters):
     """Analyze MARL policies from a saved policy through an analysis script"""
     sim, trainer = _start(full_trained_directory, parameters.checkpoint, parameters.seed)
 
+    # The sim may be wrapped by an external wrapper, which we support, but we need
+    # to unwrap it.
+    if not isinstance(sim, SimulationManager):
```

Comment:
Enforce the interfaces that we support. Checks for SimulationManager and ExternalWrapper types

Reply:
This should not be caught at analysis time. The train config api specifies that the simulation should be of type SimulationManager or Wrapper. If that is not followed, then we have undefined behavior.